dali/pipeline/executor/executor.h

```diff
@@ -659,15 +706,29 @@ void Executor<WorkspacePolicy, QueuePolicy>::PresizeData(
               tensor_to_store_queue[tensor.id]);
           for (auto storage : queue) {
             if (hint && storage->is_pinned()) {
-              storage->reserve(hint, batch_size_);
+              if (node.op->CanInferOutputs()) {
```

Comment:
Maybe:
```
          auto& queue = get_queue<op_type_static, StorageDevice::CPU>(
              tensor_to_store_queue[tensor.id]);
          for (auto storage : queue) {
            if (hint && storage->is_pinned()) {
              if (node.op->CanInferOutputs()) {
                storage->reserve(hint * batch_size_);
              } else {
                storage->reserve(hint, batch_size_);
              }
            }
            if (node.op->CanInferOutputs()) {
              storage->SetContiguous(true);
            }
          }
```
can be extracted to some template function depending on StorageDevice type.

Reply:
If you don't think it deserves a full-fledged function, create a lambda for it.

Reply:
I tried to simplify it with VALUE_SWITCH, and two generic lambdas. Doesn't sound that good but removed the code duplication.